var/spack/repos/builtin/packages/reframe/package.py

```diff
@@ -22,13 +23,38 @@ class Reframe(Package):
     maintainers = ['victorusu', 'vkarak']
 
     version('master',   branch='master')
+    version('3.0',       sha256='c4fe84a92d961546e4d0e33ca3476ba0d4cebc908eb4e33897f646bd1fd5205b')
     version('2.21',      sha256='f35d4fda2f9672c87d3ef664d9a2d6eb0c01c88218a31772a6645c32c8934c4d')
     version('2.20',      sha256='310c18d705858bbe6bd9a2dc4d382b254c1f093b0671d72363f2111e8c162ba4')
     version('2.17.3',    sha256='dc8dfb2ccb9a966303879b7cdcd188c47063e9b7999cbd5d6255223b066bf357')
     version('2.17.2',    sha256='092241cdc15918040aacb922c806aecb59c5bdc3ff7db034a4f355d39aecc101')
     version('2.17.1',    sha256='0b0d32a892607840a7d668f5dcea6f03f7022a26b23e5042a0faf5b8c41cb146')
 
-    depends_on('python@3.5:', type=('run'))
+    variant("docs", default=False,
+            description="Build ReFrame's man page documentation")
+    variant("gelf", default=False,
+            description="Add graylog handler support")
+
+    depends_on('python@3.5:', when=('@2.0:2.999'), type=('run'))
+    depends_on('python@3.6:', when=('@3.0:'), type=('run'))
+    depends_on('py-jsonschema', type=('build', 'run'))
+    depends_on('py-setuptools', type=('build', 'run'))
```

Comment:
Are you sure setuptools is needed at run-time?

Reply:
Yes. ReFrame requires it.